tools/swift-inspect/Sources/swift-inspect/WindowsRemoteProcess.swift

```diff
@@ -233,37 +236,263 @@ internal final class WindowsRemoteProcess: RemoteProcess {
       return
     }
-    defer { CloseHandle(hSnapshot) }
 
-    var heap: HEAPLIST32 = HEAPLIST32()
-    heap.dwSize = SIZE_T(MemoryLayout<HEAPLIST32>.size)
-    if !Heap32ListFirst(hSnapshot, &heap) {
-      // FIXME(compnerd) log error
+    // The size of the shared memory buffer and the names of shared
+    // memory and event objects
+    let bufSize = 512
+    let sharedMemoryName = "Local\\SwiftInspectFileMapping"
+    let readEventName = "Local\\SwiftInspectReadEvent"
+    let writeEventName = "Local\\SwiftInspectWriteEvent"
+    let waitTimeoutMs = DWORD(30000)
+
+    // Set up the shared memory
+    let hMapFile = CreateFileMappingA(INVALID_HANDLE_VALUE,
+                                      UnsafeMutablePointer(bitPattern: 0),
+                                      DWORD(PAGE_READWRITE),
+                                      0,
+                                      DWORD(bufSize),
+                                      sharedMemoryName);
+    if hMapFile == HANDLE(bitPattern: 0) {
+      print("CreateFileMapping failed \(GetLastError())")
+      return
+    }
+    if GetLastError() == ERROR_ALREADY_EXISTS {
+      print("FileMapping in use. Try again in a while.")
+      CloseHandle(hMapFile)
+      return
+    }
+    defer { CloseHandle(hMapFile) }
+    let FILE_MAP_ALL_ACCESS = STANDARD_RIGHTS_REQUIRED | SECTION_QUERY |
+                              SECTION_MAP_WRITE | SECTION_MAP_READ |
+                              SECTION_MAP_EXECUTE|SECTION_EXTEND_SIZE
+    let buf: LPVOID = MapViewOfFile(hMapFile,
+                                    DWORD(FILE_MAP_ALL_ACCESS),
+                                    0,
+                                    0,
+                                    SIZE_T(bufSize));
+    if buf == LPVOID(bitPattern: 0) {
+      print("MapViewOfFile failed \(GetLastError())")
+      return
+    }
+    defer { UnmapViewOfFile(buf) }
+
+    // Set up the event objects
+    let hReadEvent: HANDLE = CreateEventA(LPSECURITY_ATTRIBUTES(bitPattern: 0),
+                                          false, // Auto-reset
+                                          false, // Initial state is nonsignaled
+                                          readEventName)
+    if hReadEvent == HANDLE(bitPattern: 0) {
+      print("CreateEvent failed \(GetLastError())")
+      return
+    }
+    defer { CloseHandle(hReadEvent) }
+    let hWriteEvent: HANDLE = CreateEventA(LPSECURITY_ATTRIBUTES(bitPattern: 0),
+                                           false, // Auto-reset
+                                           false, // Initial state is nonsignaled
+                                           writeEventName)
+    if hWriteEvent == HANDLE(bitPattern: 0) {
+      print("CreateEvent failed \(GetLastError())")
+      return
+    }
+    defer { CloseHandle(hWriteEvent) }
+
+    // The path to the dll assuming it's in the same directory as swift-inspect.
+    let dllPath: String = Bundle.main.bundlePath + "\\SwiftInspectHeapWalk.dll"
```

Comment:
Hmm, if we use the bundle, that makes deploying this a bit more difficult as that has requirements on the layout.